src/ahbicht/validation/validation.py

```diff
@@ -0,0 +1,357 @@
+                    segment_group_validation.requirement_validation,
+                    soll_is_required,
+                )
+            )
+
+    validation_results_of_children = await asyncio.gather(*tasks)
+
+    for sublist in validation_results_of_children:
+        for item in sublist:
+            validation_results_in_context.append(item)
+
+    return validation_results_in_context
+
+
+async def validate_segment(
+    segment: Segment,
+    segment_group_requirement: Optional[RequirementValidationValue] = None,
+    soll_is_required: bool = True,
+) -> List[ValidationResultInContext]:
+    """
+    Validates a segment and the data elements it contains.
+    :param segment: the segment that should be validated
+    :param segment_group_requirement: the requirement of the segment's segment_group, e.g. IS_REQUIRED
+    :param soll_is_required: true (default) if SOLL should be handled like MUSS, false if it should be handled like KANN
+    :return: List of ValidationResultInContext of the segment and the data elements it contains
+    """
+
+    # validation of this segment
+    if segment_group_requirement is RequirementValidationValue.IS_FORBIDDEN:
+        segment_validation = SegmentLevelValidationResult(
+            requirement_validation=RequirementValidationValue.IS_FORBIDDEN
+        )
+    else:
+        segment_validation = await get_segment_level_requirement_validation_value(
+            segment, segment_group_requirement, soll_is_required
+        )
+
+    validation_results_in_context = [
+        ValidationResultInContext(discriminator=segment.discriminator, validation_result=segment_validation)
+    ]
+
+    tasks = []
+
+    # validation of this segments dataelements
+    for dataelement in segment.data_elements:
+        tasks.append(validate_dataelement(dataelement, segment_validation.requirement_validation))
+
+    validation_results_in_context_dataelements = await asyncio.gather(*tasks)
+
+    return [*validation_results_in_context, *validation_results_in_context_dataelements]
+
+
+async def get_segment_level_requirement_validation_value(
+    segment_level: SegmentLevel,
+    parent_segment_group_requirement: Optional[RequirementValidationValue] = None,
+    soll_is_required: bool = True,
+) -> SegmentLevelValidationResult:
+    """
+    Parses and evaluates a segment level expression and maps it to a requirement validation value,
+    also depending on a parent segment group requirement if present.
+    :param segment_level: the segment or segment group that should be validated
+    :param parent_segment_group_requirement: the requirement of the segment level's segment group, e.g. IS_REQUIRED
+    :param soll_is_required: true (default) if SOLL should be handled like MUSS, false if it should be handled like KANN
+    :return: Validation Result of the data element
+    """
+
+    expression_tree = parse_ahb_expression_to_single_requirement_indicator_expressions(segment_level.ahb_expression)
```

Comment:
sind wir sicher, dass die ahb_expression nie empty oder None ist? denn mindestens empty würde ja einen fehler werfen glaube ich

Reply:
In der Maus sollte es (in Zukunft issue: https://github.com/Hochfrequenz/mig_ahb_utility_stack/issues/38) nicht mehr möglich sein.

Reply:
Ein Check ist immer gut, dann könnte man es auf "Forbidden" stellen.

Reply:
Hm, ich hab noch mal nachgedacht, ist es nicht sogar richtig, dass hier ein Fehler geworfen wird?

Reply:
Wir haben beschlossen, dass es richtig ist, dass hier ein Fehler geworfen wird.